### PR TITLE
CrossPlatformSocket: Added SocketOptionName.ReuseAddress = true.

### PR DESCRIPTION
**Pull request by SeppPenner**
Fixes https://github.com/chkr1011/MQTTnet/issues/494 (At least as far as I understood it).
Suggestion from @zvrba.

Might also fix https://github.com/chkr1011/MQTTnet/issues/872.

### CODE REVIEW

File: Source/MQTTnet/Implementations/CrossPlatformSocket.cs [hunk 1]
```diff
@@ -86,6 +86,7 @@ public void Bind(EndPoint localEndPoint)
         {
             if (localEndPoint is null) throw new ArgumentNullException(nameof(localEndPoint));
 
+            _socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, true);
```
Comment (chkr1011) [inline Source/MQTTnet/Implementations/CrossPlatformSocket.cs:89]: I don't think we need this. There is already a property which sets the socket option. The property is also set when the server starts. Probably the user who has problems with that only needs to enable this in the server options.

Comment (SeppPenner) [reply Source/MQTTnet/Implementations/CrossPlatformSocket.cs:89]: Ok, can you tell him how to enable this, please? https://github.com/chkr1011/MQTTnet/issues/872.

I will close the pull request soon.

Comment (SeppPenner) [reply Source/MQTTnet/Implementations/CrossPlatformSocket.cs:89]: Will be closed as I have given a hint to the original person who had this problem.

**Comment by SeppPenner** [comment]
Not needed any longer, see https://github.com/chkr1011/MQTTnet/issues/872#issuecomment-644053219 for more information.

**Pull request closed without merging**